src/PriceCalculatorKata/Product.cs

```diff
@@ -40,11 +42,30 @@ public void CheckPercentageValidation(double value)
         }
         public void ReportProductPrice()
         {
-            double DiscountAmount = Math.Round(Price * DiscountPercentage, 2);
-            double PriceWithTaxAndDiscount = Math.Round(Price * (1 + TaxPercentage - DiscountPercentage), 2);
-            Console.WriteLine($"${Price} before tax and discount and ${PriceWithTaxAndDiscount} after {TaxPercentage * 100}% tax and {DiscountPercentage * 100}% discount");
+            double DiscountAmount = GetTotalDiscountAmount();
+            double TaxAmount = Price * TaxPercentage;
+            double PriceWithTaxAndDiscount = Math.Round(Price + TaxAmount - DiscountAmount, 2);
+            Console.WriteLine($"${Price} before tax and discount and ${PriceWithTaxAndDiscount} after {TaxPercentage * 100}% tax and {UniversalDiscountPercentage * 100}% discount");
             if (DiscountAmount > 0)
-                Console.WriteLine($"Discount amount = {DiscountAmount}");
+                Console.WriteLine($"Total Discount amount = {DiscountAmount}");
+        }
+        private double GetTotalDiscountAmount()
+        {
+            double UnivarsalDiscountAmount = Math.Round(Price * UniversalDiscountPercentage, 2);
+            Console.WriteLine($"Univarsal Discount amount = {UnivarsalDiscountAmount}");
+            switch (UPC)
+            {
+                case 12345:
```

Comment:
hmmm this is not readable really, and why we are hardcoding this here. Would be better to make this more generic. I know in the requirement there's an example, but we want to make of course more generic. Think about a better design for this, where we can hold the discount for each Product/UPC ;)